pageserver/src/tenant.rs

```diff
@@ -5119,9 +5149,6 @@ mod tests {
                 )
                 .await?;
             tline.freeze_and_flush().await?;
-            tline
-                .compact(&CancellationToken::new(), EnumSet::empty(), &ctx)
```

Comment:
is this removal of compaction expected?

Reply:
... and I don't see any other compactions running in this test case, so I assume this is rather unexpected?

Reply:
It's indeed expected: compaction removes those keys. This sounds scary because the legacy compaction didn't do that: but that's because it never created image layers. the bug is more in collect_keyspace, used both by new and old compaction, returning key ranges that don't contain the written key. So that's why I'm removing compaction here.

Same issue in the other test as well.